tests/std/tests/P0896R4_ranges_alg_copy_n/test.cpp

```diff
@@ -9,34 +9,27 @@
 
 #include <range_algorithm_support.hpp>
 
-constexpr void smoke_test() {
-    using ranges::copy_n, ranges::copy_n_result, ranges::iterator_t;
-    using std::same_as;
+using namespace std;
+using same_as;
+using ranges::copy_n, ranges::copy_n_result, ranges::iterator_t;
 
-    // Validate that copy_n_result aliases in_out_result
-    STATIC_ASSERT(same_as<copy_n_result<int, double>, ranges::in_out_result<int, double>>);
-
-    int const input[] = {13, 42, 1729};
-    int output[]      = {-1, -1, -1};
-    basic_borrowed_range wrapped_input{input};
-    auto result = copy_n(wrapped_input.begin(), ranges::distance(input), basic_borrowed_range{output}.begin());
-    STATIC_ASSERT(same_as<decltype(result),
-        copy_n_result<iterator_t<basic_borrowed_range<int const>>, iterator_t<basic_borrowed_range<int>>>>);
-    assert(result.in == wrapped_input.end());
-    assert(result.out == basic_borrowed_range{output}.end());
-    assert(ranges::equal(output, input));
-}
-
-int main() {
-    STATIC_ASSERT((smoke_test(), true));
-    smoke_test();
-}
+// Validate that copy_n_result aliases in_out_result
+STATIC_ASSERT(same_as<copy_n_result<int, double>, ranges::in_out_result<int, double>>);
 
 struct instantiator {
-    template <class In, class, class Out>
-    static void call(In in = {}, std::iter_difference_t<In> const count = 42, Out out = {}) {
-        (void) ranges::copy_n(std::move(in), count, std::move(out));
+    static constexpr int input[3] = {13, 42, 1729};
+    template <class In, class, class Write>
+    static constexpr void call() {
+        int output[3] = {-1, -1, -1};
+        auto result   = copy_n(In{input}, ranges::distance(input), Write{output});
+        STATIC_ASSERT(same_as<decltype(result), copy_n_result<In, Write>>);
+        assert(result.in.base() == input + 3);
+        assert(result.out.base() == output + 3);

```

Comment:
Ugh: I lied. When these are single-pass iterators you need to call `base` on an rvalue until #959 is applied.
```suggestion
        assert(move(result.in).base() == input + 3);
        assert(move(result.out).base() == output + 3);
```

Reply:
Gasp, actually something tingled in my mind because I had just read through the iterator class. 

But that sweet little "commit suggestion" button is only a click away

Reply:
I decided to go forward and prepare the PR so that i only need to rebase once the machinery has landed